packages/core/src/components/forms/numericInputUtils.ts

```diff
@@ -77,9 +128,23 @@ export function isValidNumericKeyboardEvent(e: React.KeyboardEvent) {
  * See here for the input[type="number"].value spec:
  * https://www.w3.org/TR/2012/WD-html-markup-20120329/input.number.html#input.number.attrs.value
  */
-const FLOATING_POINT_NUMBER_CHARACTER_REGEX = /^[Ee0-9\+\-\.]$/;
-function isFloatingPointNumericCharacter(character: string) {
-    return FLOATING_POINT_NUMBER_CHARACTER_REGEX.test(character);
+function isFloatingPointNumericCharacter(character: string, locale: string | undefined) {
+    if (locale) {
```

Comment:
prefer explicit boolean expressions rather than implicit type coercion

```suggestion
    if (locale !== undefined) {
```